redbot/setup.py

```diff
@@ -262,13 +268,13 @@ async def remove_instance(
     if _create_backup is True:
         await create_backup(instance)
 
-    backend = get_current_backend(instance)
-    driver_cls = drivers.get_driver_class(backend)
-    await driver_cls.initialize(**data_manager.storage_details())
-    try:
-        if delete_data is True:
+    driver_cls = drivers.get_driver_class(backend)  
+    if delete_data is True:
+        try:
+            await driver_cls.initialize(**data_manager.storage_details())
             await driver_cls.delete_all_data(interactive=interactive, drop_db=drop_db)
-
+        finally:
+            await driver_cls.teardown()
         if interactive is True and remove_datapath is None:
```

Comment:
This and the following lines should not be part of the `if delete_date is True` block

Reply:
Hopefully done!